wxgui-src/wxmain.cc

```diff
@@ -615,6 +664,7 @@ void GameApp::undo_move(wxCommandEvent &)
         GalaxyWidget *gp = (GalaxyWidget *)wxWindow::FindWindowById(wxID_GALAXY_MAP);
         assert(gp != NULL);
         gp->update(this->history.currentstate());
+        save_game();
         global_attacker = this->history.current_attacker();
```

Comment:
(A) Consider refactoring lines 664–668 into a function called something like `GameApp::update_state()`.

(B) The autosave functionality, if you keep it, should record *all* of history, followed by the appropriate number of `undo` commands. If you've just undone a move, you should be able to "save, restore, redo"; the game shouldn't forget your "future history" just because you saved and restored.